src/main/java/Ping/Parser.java

```diff
@@ -0,0 +1,196 @@
+package Ping;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import Command.AddCommand;
+import Command.BlahCommand;
+import Command.Command;
+import Command.DeleteCommand;
+import Command.ExitCommand;
+import Command.FindCommand;
+import Command.HiCommand;
+import Command.ListCommand;
+import Command.MarkCommand;
+import Command.UnmarkCommand;
+/**
+ * This class is used to parse the command input by the user
+ */
+public class Parser {
+    /**
+    * This method is used to parse the command input by the user
+     * @param input the command input by the user
+     * @return the command object
+     */
+    public static Command parseCommand(TaskList tasks, String input) throws PingException {
+        String[] restCommands = input.split(" ");
+        String command = restCommands[0];
+        switch (command.toLowerCase()) {
+        case "hi":
+            return new HiCommand();
+        case "blah":
+            return new BlahCommand();
+        case "bye":
+            return new ExitCommand();
+        case "mark":
+            return parseMark(restCommands);
+        case "unmark":
+            return parseUnmark(restCommands);
+        case "todo":
+            return parseTodo(restCommands);
+        case "list":
+            return new ListCommand();
+        case "delete":
+            return parseDelete(restCommands);
+        case "deadline":
+            return parseDeadline(restCommands);
+        case "event":
+            return parseEvent(restCommands);
+        case "find":
+            return parseFind(restCommands);
```

Comment:
I like how you seperated the parse methods into private methods in this class, makes it cleaner instead of one long method!